frontend/public/components/utils/cog.jsx

```diff
@@ -2,27 +2,51 @@ import * as _ from 'lodash-es';
 import * as React from 'react';
 import * as classNames from 'classnames';
 import { Tooltip } from './tooltip';
+import * as PropTypes from 'prop-types';
 
 import { annotationsModal, configureReplicaCountModal, labelsModal, nodeSelectorModal, podSelectorModal, deleteModal } from '../modals';
 import { DropdownMixin } from './dropdown';
 import { history, resourceObjPath } from './index';
-import { referenceForModel } from '../../module/k8s';
-import { connectToModel } from '../../kinds';
+import { referenceForModel, kindForReference } from '../../module/k8s';
+import { kindReducerName } from '../../kinds';
+import store from '../../redux';
+
 
 const CogItems = ({options, onClick}) => {
   const visibleOptions = _.reject(options, o => _.get(o, 'hidden', false));
-  const lis = _.map(visibleOptions, (o, i) => <li key={i}><a onClick={e =>onClick(e, o)}>{o.label}</a></li>);
+  const lis = _.map(visibleOptions, (o, i) => <li key={i}><a onClick={e => onClick(e, o)}>{o.label}</a></li>);
   return <ul className="dropdown-menu co-m-cog__dropdown">
     {lis}
   </ul>;
 };
 
+
 export class Cog extends DropdownMixin {
   constructor(props) {
     super(props);
     this.onClick = (...args) => this.onClick_(...args);
   }
 
+  componentWillReceiveProps () {
+    // Stop call to parent for better performance
+    return;
+  }
+
+  calculateOptions () {
+    const {actions, kind, options, resource} = this.props;
+    if (options) {
+      return options;
+    }
+    // Latest possible binding to the Store for better mounting times
+    //  NOTE: binding and passing this function in from ResourceCog resulted in ~25% worse performance when mounting, hence the crazy interface
```

Comment:
Nice, how did you measure / have a quick screenshot of perf diff?  No worries if no, just curious.